### ISSUE TITLE
Inference of tuple default values too strict after 1.1.224

### ISSUE THREAD

**Issue by bellini666** [microsoft/pyright#3107]
After updating to 1.1.224 I just notice that a lot of errors when using some functions from some libs that looks like this:

```python
def some_func(list_of_args=()):
    ...
```

Pyright is inferring that `list_of_args` to of type of a tuple with no elements, which prevents me from passing anything other than an empty tuple to it. For example:

```python
some_func(["something"])  # "list[str]" is incompatible with "tuple[()]" (reportGeneralTypeIssues)
some_func(("something",))  # Element size mismatch; expected 0 but received 1 (reportGeneralTypeIssues)
```

I know that this is mostly the libs fault, but a lot of libs, specially old ones, liked to use tuples as default "empty list" values, and they are usually not correctly typed. I don't think the use case is correct (tuples positions and size have meaning, lists have not), but that's how a lot of them does. Django for example is one of those who uses tuples extensively to define things that should actually be lists, or even as like this example here.

Is there a way to avoid that inference when the argument type is a tuple, or at least when it is an empty tuple?

**Comment by erictraut** [comment]
Yes, I agree that parameter type inference should not be used when the default argument is a tuple, list, set or dict expression. The change will be in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.225, which I just published. It will also be included in the next release of pylance.
